src/WebJobs.Script/Binding/ServiceBusScriptBindingProvider.cs

```diff
@@ -110,8 +112,17 @@ public override Type DefaultType
                 {
                     if (Context.Access == FileAccess.Read)
                     {
-                        return string.Compare("binary", Context.DataType, StringComparison.OrdinalIgnoreCase) == 0
+                        Type type = string.Compare("binary", Context.DataType, StringComparison.OrdinalIgnoreCase) == 0
                             ? typeof(byte[]) : typeof(string);
+
+                        if (string.Compare("many", Context.Cardinality, StringComparison.OrdinalIgnoreCase) == 0)
+                        {
+                            // arrays are supported for both trigger input as well
+                            // as output bindings
+                            type = type.MakeArrayType();
+                        }
```

Comment:
I agree. I've opened https://github.com/Azure/azure-webjobs-sdk-script/issues/737 to address this separately, since it will require more thought. E.g. cardinality "many" might not map to an array for some binding, but a different collection type. So for now we rely on the extension to specify this explicitly.